packages/trader/src/App/Components/Elements/Modals/ServicesErrorModal/__tests__/account-verification-required-modal.spec.tsx

```diff
@@ -0,0 +1,93 @@
+import { Router } from 'react-router-dom';
+import userEvent from '@testing-library/user-event';
+import { routes, isMobile } from '@deriv/shared';
+
+type TModal = React.FC<{
+    children: React.ReactNode;
+    is_open: boolean;
+    title: string;
+    height: string;
+}> & {
+    Body?: React.FC<{
+        children: React.ReactNode;
+    }>;
+    Footer?: React.FC<{
+        children: React.ReactNode;
+    }>;
+};
+
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+    isMobile: jest.fn(() => true),
+}));
+
+jest.mock('@deriv/components', () => {
+    const original_module = jest.requireActual('@deriv/components');
+    const Modal: TModal = jest.fn(({ children, is_open, title, height }) => {
+        if (is_open) {
+            return (
+                <div data-testid='modal'>
+                    <h3>{title}</h3>
+                    <p>{height}</p>
+                    {children}
+                </div>
+            );
+        }
+        return null;
+    });
+    Modal.Body = jest.fn(({ children }) => <div>{children}</div>);
+    Modal.Footer = jest.fn(({ children }) => <div>{children}</div>);
+
+    return {
+        ...original_module,
+        Modal,
+    };
+});
+describe('<AccountVerificationRequiredModal />', () => {
+    let mocked_props: React.ComponentProps<typeof AccountVerificationRequiredModal>;
+    const history = createBrowserHistory();
+    const renderWithRouter = (component: React.ReactElement) => {
+        return render(<Router history={history}>{component}</Router>);
+    };
+
+    beforeEach(() => {
+        mocked_props = {
+            onConfirm: jest.fn(),
+            is_visible: true,
+        };
+    });
+    it('height should be auto if isMobile is true', () => {
+        render(<AccountVerificationRequiredModal {...mocked_props} />);
+        expect(screen.getByText('auto')).toBeInTheDocument();
+    });
+    it('height should be 220px if isMobile is false', () => {
+        (isMobile as jest.Mock).mockReturnValue(false);
+        render(<AccountVerificationRequiredModal {...mocked_props} />);
+        expect(screen.getByText('220px')).toBeInTheDocument();
+    });
+    it('should render modal title, modal description, and submit button.', () => {
+        render(<AccountVerificationRequiredModal {...mocked_props} />);
+        expect(screen.getByText(/account verification required/i)).toBeInTheDocument();
+        expect(
+            screen.getByText(
+                /Please submit your proof of identity and proof of address to verify your account and continue trading./i
+            )
+        ).toBeInTheDocument();
+        expect(screen.getByText(/submit proof/i)).toBeInTheDocument();
+    });
+    it('should return null when is_visible is false', () => {
+        mocked_props.is_visible = false;
+        const { container } = render(<AccountVerificationRequiredModal {...mocked_props} />);
+        expect(container).toBeEmptyDOMElement();
```

Comment:
```suggestion
        expect(container).toBeNull();
```
Thinking about whether this assertion be `right`? 
Like the modal component might still be present in the DOM tree, but hidden due to CSS styles or other mechanisms.
Checking for Null might be a better approach here.

Reply:
AFAIK. Modal doesn't get injected into the DOM until its open and unmounted on the exit.

container returns < div />. What do you think?

Reply:
https://reactcommunity.org/react-transition-group/css-transition

Check example provided in the link. Modal doesnt get injected until "Show Message" is clicked and Modal gets removed on close.